WordPress/gutenberg
# Block Supports: Add text columns (column count) to typography block supports

Related: 
- https://github.com/WordPress/gutenberg/pull/46566
- https://github.com/WordPress/gutenberg/issues/25459 
- https://github.com/WordPress/gutenberg/issues/27118

## What?

Adds a new block support feature to the typography toolset to allow control over CSS column count for blocks displaying multiline text.

## Why?

To open up further design possibilities.

### Patterns enabled by this

| Post Excerpt | Paragraph |
|---|---|
| <img src="https://user-images.githubusercontent.com/908665/93613659-0f615080-f9c9-11ea-8480-94d80ee6d2ad.png" width="400"/> | <img src="https://user-images.githubusercontent.com/275961/99705247-60f18e80-2a91-11eb-8775-08020ef0a167.png" width="400" /> |


## How?
- Adds new "text columns" block support that maps to the `column-count` CSS property.
- The new block support and design tool is disabled by default (happy to change this if desired)
- Updates global styles typography panel to include new control if enabled.

## Testing Instructions

1. Run `npm run test:unit:php -- --filter WP_Theme_JSON_Gutenberg_Test`
2. Run `npm run test:unit packages/block-editor/src/hooks/test/style.js`
3. Enable the text columns UI by setting your theme.json's `settings.typography.textColumns` to `true`
4. Opt into text columns support for a block such as Post Excerpt by setting its `supports.typography.textColumns` property to `true`.
5. Edit a post, add your block with text columns support, add text to it, then select it.
6. In the typography panel in the sidebar, toggle on the text columns control via the panel's ellipsis menu
7. Adjust the control and ensure it works.
8. Save the post and confirm the columns display appropriately on the frontend.
9. Test text columns in global styles for blocks works.
10. Make sure the text columns control isn't an option under the Global Stlyes' root typography panel.

## Screenshots 


https://user-images.githubusercontent.com/60436221/207778088-553f8e08-6b0c-48f6-b432-e1ba6334f854.mp4
